### PR TITLE
fix: [CDS-73583]: added support of toggle in dynamic provisioner

### PR DESCRIPTION
**Pull request by satyam-harness**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by pankajsingh-harness — DISMISSED** [review]

File: src/modules/70-pipeline/components/PipelineStudio/ExecutionGraph/ExecutionGraph.tsx [hunk 1]
```diff
@@ -826,15 +829,15 @@ function ExecutionGraphRef<T extends StageElementConfig>(
           templateIcons,
           serviceDependencies: undefined,
           errorMap: errorMap,
-          parentPath: `${stagePath}.stage.spec.execution.rollbackSteps`
+          parentPath: `${stagePath}.${getParentPath(isProvisioner)}.rollbackSteps`
```
Comment (PratyushGar) [inline src/modules/70-pipeline/components/PipelineStudio/ExecutionGraph/ExecutionGraph.tsx:832]: declare this above in a separate variable to be used in both places

**Review by PratyushGar — DISMISSED** [review]

**Comment by satyam-harness** [comment]
retrigger titlecheck

**Review by udayvunnam — DISMISSED** [review]

**Review by PratyushGar — APPROVED** [review]

File: src/modules/70-pipeline/components/PipelineDiagram/Nodes/DefaultNode/PipelineStepNode/PipelineStepNode.tsx [hunk 2]
```diff
@@ -354,10 +359,10 @@ function PipelineStepNode(props: PipelineStepNodeProps): JSX.Element {
         )}
         {CODE_ICON && (
           <Icon
-            className={defaultCss.codeIcon}
             color={isSelectedNode() ? Color.WHITE : undefined}
             size={8}
             name={CODE_ICON}
+            {...(whenCondition ? { className: defaultCss.disabledIcon } : { className: defaultCss.codeIcon })}
```
Comment (pankajsingh-harness) [inline src/modules/70-pipeline/components/PipelineDiagram/Nodes/DefaultNode/PipelineStepNode/PipelineStepNode.tsx:365]:
```suggestion
            className = { whenCondition ? defaultCss.disabledIcon : defaultCss.codeIcon }
```

**Review by udayvunnam — APPROVED** [review]

File: src/modules/70-pipeline/components/PipelineDiagram/Nodes/DefaultNode/PipelineStepNode/PipelineStepNode.tsx (hunk 2 above)
Comment (satyam-harness) [reply src/modules/70-pipeline/components/PipelineDiagram/Nodes/DefaultNode/PipelineStepNode/PipelineStepNode.tsx:365]: i can fix it in subsequent PR

**Pull request merged**